filters/auth/jwt_validation.go

```diff
@@ -0,0 +1,170 @@
+package auth
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/MicahParks/keyfunc"
+	"github.com/golang-jwt/jwt"
+	"github.com/opentracing/opentracing-go"
+	log "github.com/sirupsen/logrus"
+	"github.com/zalando/skipper/filters"
+)
+
+const (
+	JwtValidationName = "jwtValidation"
+)
+
+type (
+	jwtValidationSpec struct {
+		options TokenintrospectionOptions
+	}
+
+	jwtValidationFilter struct {
+		jwksUri string
+	}
+)
+
+var jwksMap map[string]*keyfunc.JWKs = make(map[string]*keyfunc.JWKs)
+
+func NewJwtValidationWithOptions(o TokenintrospectionOptions) filters.Spec {
+	return &jwtValidationSpec{
+		options: o,
+	}
+}
+
+func NewJwtValidation(timeout time.Duration) filters.Spec {
+	return NewJwtValidationWithOptions(TokenintrospectionOptions{
+		Timeout: timeout,
+		Tracer:  opentracing.NoopTracer{},
+	})
+}
+
+func JwtValidationWithOptions(
+	create func(time.Duration) filters.Spec,
+	o TokenintrospectionOptions,
+) filters.Spec {
+	s := create(o.Timeout)
```

Comment:
I can not find this `create` function

Reply:
It is here https://github.com/abinet/skipper/blob/master/skipper.go#L1235
And this is the reason why second constructor is being needed.

Reply:
I don't get it. Lets have single constructor with options, timeouts etc. Tests can introduce helper method to use whatever timeout or NoopTracer needed.

Reply:
I.e. have only `NewJwtValidationWithOptions`